src/v/pandaproxy/json/requests/error_reply.h

```diff
@@ -20,10 +22,21 @@
 namespace pandaproxy::json {
 
 struct error_body {
-    ss::httpd::reply::status_type error_code;
+    json::error_code error_code;
```

Comment:
should we rename `error_code` as `ec`. sometimes is hard to read variable names as types. but just a nit, up to you.

Reply:
Done, it's also now a `std::error_condition`.